src/components/Sidebar.js

```diff
@@ -1,13 +1,72 @@
-import { Link } from 'react-router-dom';
+import { useState } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import { Icon } from '@iconify/react';
 
 export default function Sidebar() {
+  const [open, setOpen] = useState(true);
+
   return (
-    <div className="w-60 p-4 flex-shrink-0 shadow-xl">
-      <h1 className="font-semibold text-2xl mb-8">Coverage Checker</h1>
-      <nav className="flex flex-col gap-2">
-        <Link to="/file-upload">Upload Files</Link>
-        <Link to="/">mCODE Coverage</Link>
-      </nav>
+    <div className={`flex flex-col ${open ? 'w-60' : 'w-20'} p-4 h-screen sticky top-0 flex-shrink-0 shadow-xl`}>
+      {/* sidebar header */}
+      <div className="flex flex-row">
+        <h1 className={`${open ? '' : 'hidden'} font-semibold text-xl mb-8`}>Coverage Checker</h1>
```

Comment:
Let's move the `mb-8` from the header to the container with the heading and the arrow icon.